client/src/components/common/PendingSaving.jsx

```diff
@@ -0,0 +1,65 @@
+import { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { SAVE_COUNTDOWN_INTERVAL, SAVE_INITIAL_TIMEOUT } from '../../constants/time';
+
+export default function PendingSaving(props) {
```

Comment:
Is `PendingSaving` a hook or a component? It looks like a component but it doesn't return any jsx?

Reply:
it's a hook designed to be composed

any ideas to improve the syntax so it looks like a hook?

Reply:
To make it a hook just rename it to `usePendingSaving`

Reply:
done :)